src/main/java/data/config/BaseData.java

```diff
@@ -6,22 +6,7 @@
 public class BaseData {
 
 
-    private static Loader loader;
```

Comment:
I think it makes sense to completely remove all pipeline specific and use just the local Implementation